skytemple_files/common/ppmdu_config/pmdsky_debug/loader.py

```diff
@@ -0,0 +1,308 @@
+        return newobj
+
+
+def _read_symbol(loadaddr: int, symbol_def: dict, typ: Pmd2BinarySymbolType, region: str) -> Optional[Pmd2BinarySymbol]:
+    assert isinstance(loadaddr, int)
+    if 'name' in symbol_def:
+        name = symbol_def['name']
+    else:
+        raise ValueError("Symbol is missing it's name.")
+    
+    if 'address' in symbol_def:
+        if region in symbol_def['address']:
+            if isinstance(symbol_def['address'][region], list):
+                begin = symbol_def['address'][region][0] - loadaddr
+            else:
+                begin = symbol_def['address'][region] - loadaddr
+        else:
+            return None
+    else:
+        raise ValueError(f"Symbol {name} is missing an address.")
+    
+    if 'description' in symbol_def:
+        description = symbol_def['description']
+    else:
+        description = ""
+    
+    if typ == Pmd2BinarySymbolType.FUNCTION:
+        return Pmd2BinarySymbol(
+            name=name,
+            typ=Pmd2BinarySymbolType.FUNCTION,
+            begin=begin,
+            end=None,
+            description=description
+        )
+    else:
+        if 'length' in symbol_def:
+            if region in symbol_def['length']:
+                end = begin + symbol_def['length'][region]
+            else:
+                return None
+        else:
+            # We assume it has length 4 (pointer length) if not defined.
+            end = begin + 4
+        
+        return Pmd2BinarySymbol(
+            name=name,
+            typ=typ,
+            begin=begin,
+            end=end,
+            description=description
+        )
+
+
+def _read(binaries: Dict[str, _IncompleteBinary], yml: dict, region: str):
+    # arm9.bin and the overlays MUST be read first.
+    for bin_name, definition in yml.items():
+        assert isinstance(bin_name, str)
+        binary = _IncompleteBinary.get(binaries, bin_name)
+        if 'address' in definition:
+            if region in definition['address']:
+                binary.loadaddress = int(definition['address'][region])
+        if 'length' in definition:
+            if region in definition['length']:
+                binary.length = int(definition['length'][region])
+        if 'functions' in definition:
+            assert binary.loadaddress is not None
+            for symbol_def in definition['functions']:
+                sym = _read_symbol(binary.loadaddress, symbol_def, Pmd2BinarySymbolType.FUNCTION, region)
+                if sym is not None:
+                    binary.symbols.append(sym)
+        if 'data' in definition:
+            assert binary.loadaddress is not None
+            for symbol_def in definition['data']:
+                sym = _read_symbol(binary.loadaddress, symbol_def, Pmd2BinarySymbolType.DATA, region)
+                if sym is not None:
+                    binary.symbols.append(sym)
+        if 'description' in definition:
+            if region in definition['description']:
```

Comment:
Descriptions aren't region-dependent, so I think the description will probably never get set like this.

Reply:
Thanks, I missed this.